backend/src/resolvers/shout.js

```diff
@@ -0,0 +1,51 @@
+export default {
+  Mutation: {
+    shout: async (_object, params, context, _resolveInfo) => {
+      const { id, type } = params
+
+      const session = context.driver.session()
+      let transactionRes = await session.run(
+        `MATCH (node {id: $id})<-[:WROTE]-(userWritten:User), (user:User {id: $userId})
+          WHERE $type IN labels(node) AND NOT userWritten.id = $userId
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T08:53:04Z" title="Friday, April 12th 2019, 10:53:04 am +02:00">Apr 12, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,51 @@
+export default {
+  Mutation: {
+    shout: async (_object, params, context, _resolveInfo) => {
+      const { id, type } = params
+
+      const session = context.driver.session()
+      let transactionRes = await session.run(
+        `MATCH (node {id: $id})<-[:WROTE]-(userWritten:User), (user:User {id: $userId})
+          WHERE $type IN labels(node) AND NOT userWritten.id = $userId
```


A future improvement could be that you can omit the `ShoutTypeEnum`. I mean: The server knows what you can shout and what not, right? So, instead of having the type of the shouted object sent to the server, you just send the id and the server checks if the labels of the matched, shouted object intersects with a well-known set of shoutable objects (by a set of labels `['Project', 'Post', ...]`).

For now it is fine though!

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-04-12T09:01:23Z" title="Friday, April 12th 2019, 11:01:23 am +02:00">Apr 12, 2019</time>_

---


That is a concurrence between two things:
- KISS: Keep it stupid simple …
- But redundancy may be good for think and error checking: What type has the Webapp programmer thought to be shouted or followed and does this fit to what can be … on the server side?

I think you just like to KISS … 😘 !?

Reply:
> [<img alt="Lulalaby" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Lulalaby) **Authored by [Lulalaby](https://github.com/Lulalaby)**
_<time datetime="2019-04-12T15:16:47Z" title="Friday, April 12th 2019, 5:16:47 pm +02:00">Apr 12, 2019</time>_

---


😆